java/connector-node/risingwave-sink-iceberg/src/main/java/com/risingwave/connector/catalog/JniCatalogWrapper.java

```diff
@@ -145,7 +145,7 @@ public void createNamespace(String namespaceStr) {
      */
     public boolean dropTable(String tableIdentifier) {
         TableIdentifier id = TableIdentifier.parse(tableIdentifier);
-        return catalog.dropTable(id);
+        return catalog.dropTable(id, true);
```

Comment:
Add purge flag to drop table api, since S3 tables need purge flag to drop table. BTW, this can simplify iceberg engine table drop logic as well.